xemu: cosmetic rewrite

AndersonTorres opened the pull request:
Also, add myself as maintainer.

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

winterqt commented:
But why? You should probably explain what exactly about this is better than the previous version (because it seems like nothing is).

AndersonTorres commented:
__Reorder the inputs and using `install` instead of `mkdir + cp`__ count?

AndersonTorres commented:
Why to split the maintainers here? It will not affect possible bisections, after all.

AndersonTorres committed:
xemu: add meta attributes

- Get rid of nested `with` in meta
  + Because `with` itself is an antipattern, as explained in
    https://nix.dev/anti-patterns/language#with-attrset-expression
  + Because nested `with` has unintuitive behavior, as shown in
    https://github.com/NixOS/nix/issues/490
- Add meta.longDescription;
- Add meta.changelog;
- Update meta.maintainers list.

AndersonTorres commented on pkgs/applications/emulators/xemu/default.nix:
```diff
@@ -112,12 +115,12 @@ stdenv.mkDerivation rec {
-      mkdir -p $out/share/icons/hicolor/$RES/apps/
-      cp ../ui/icons/xemu_$RES.png $out/share/icons/hicolor/$RES/apps/xemu.png
-    done
+  ''
+  + (lib.concatMapStringsSep "\n"
+    (res:
+      "install -Dm644 -T ../ui/icons/xemu_${res}.png $out/share/icons/hicolor/${res}/apps/xemu.png")
+    [ "16x16" "24x24" "32x32" "48x48" "128x128" "256x256" "512x512" ])
+  + ''
```
I am a bit averse of using Bash, it is unpredictable sometimes. I prefer to treat it as a lower level language, and use higher level constructions from Nix itself to express intent.

AndersonTorres committed:
xemu: cosmetic rewrite

- Use rec-less, overlay-style overridable attributes for stdenv.mkDerivation;
- Reorder the lists and sets in asciibetical order;
- Format the code.

AndersonTorres committed:
xemu: refactor installPhase

- Use install command instead of mkdir + cp
- Use nix functional code instead of shell for loop

AndersonTorres commented:
@ofborg build xemu
